.github/actions/get-scenarios/get-scenarios.sh

```diff
@@ -0,0 +1,221 @@
+            break
+            ;;
+        *)
+            echo "ERROR: Unexpected argument [${1}]"
+            shift
+            exit 1
+            ;;
+    esac
+done
+
+case "${RUNNER_TYPE}" in
+    "github")
+        if [ -n "${RUNNER_TAGS}" ]; then
+            echo "ERROR: The 'github' runner type does not support runner tags"
+            exit 1
+        fi
+
+        echo "Runner type is 'github'"
+        ;;
+    "self")
+        echo "Runner type is 'self'"
+        ;;
+    *)
+        echo "ERROR: Unsupported runner type specified [${RUNNER_TYPE}]"
+        exit 1
+        ;;
+esac
+
+if [ -n "${BENCHMARK_QUERY}" ]; then
+    case "${BENCHMARK_QUERY}" in
+        "fio"|"uperf"|"iperf"|"oslat")
+            echo "Benchmark is '${BENCHMARK_QUERY}'"
+            ;;
+        *)
+            echo "ERROR: Unsupported benchmark specified [${BENCHMARK_QUERY}]"
+            exit 1
+            ;;
+    esac
+else
+    echo "No benchmark specified"
+fi
+
+function get_scenario() {
+    local benchmark endpoint enabled
+
+    endpoint=${1}; shift
+    benchmark=${1}; shift
+    enabled=${1}; shift
+
+    echo "{ \"endpoint\": \"${endpoint}\", \"benchmark\": \"${benchmark}\", \"enabled\": ${enabled} },"
+}
+
+function get_enabled_scenario() {
+    local benchmark endpoint
+
+    endpoint=${1}; shift
+    benchmark=${1}; shift
+
+    get_scenario "${endpoint}" "${benchmark}" "true"
+}
+
+function get_disabled_scenario() {
+    local benchmark endpoint
+
+    endpoint=${1}; shift
+    benchmark=${1}; shift
+
+    get_scenario "${endpoint}" "${benchmark}" "false"
+}
+
+function log_enabled() {
+    local benchmark endpoint
+
+    endpoint=${1}; shift
+    benchmark=${1}; shift
+
+    echo "Adding enabled scenario: endpoint=${endpoint} benchmark=${benchmark}"
+}
+
+function log_disabled() {
+    local benchmark endpoint
+
+    endpoint=${1}; shift
+    benchmark=${1}; shift
+
+    echo "Adding disabled scenario: endpoint=${endpoint} benchmark=${benchmark}"
+}
+
+scenarios_json="["
+
+case "${RUNNER_TYPE}" in
+    "github")
+        if [ -n "${BENCHMARK_QUERY}" ]; then
+            case "${BENCHMARK_QUERY}" in
+                "fio"|"uperf"|"iperf")
+                    for endpoint in "k8s" "remotehost"; do
+                        log_enabled "${endpoint}" "${BENCHMARK_QUERY}"
+                        scenarios_json+=$(get_enabled_scenario "${endpoint}" "${BENCHMARK_QUERY}")
+                    done
+                    ;;
+                "oslat")
+                    for endpoint in "k8s"; do
+                        log_enabled "${endpoint}" "${BENCHMARK_QUERY}"
+                        scenarios_json+=$(get_enabled_scenario "${endpoint}" "${BENCHMARK_QUERY}")
+                    done
+                    ;;
+            esac
+        else
+            for endpoint in "k8s" "remotehost"; do
+                case "${endpoint}" in
+                    "k8s")
+                        for benchmark in "fio" "uperf" "iperf" "oslat"; do
```

Comment:
if this list comes from a file you only need to change the file, not the code.

Reply:
I'm open to doing that in the future if/when I have a better feeling for what the various choices are.  I feel like I need more choices before I arrive at a proper configuration format.